discovery_engine_core/core/src/engine.rs

```diff
@@ -447,6 +450,31 @@ where
             Ok(documents)
         }
     }
+
+    /// Sets a new list of excluded sources
+    pub async fn set_excluded_sources(
+        &mut self,
+        history: &[HistoricDocument],
+        excluded_sources: Vec<String>,
+    ) -> Result<(), Error> {
+        let exclusion_set = excluded_sources.iter().cloned().collect::<HashSet<_>>();
+        *self.config.excluded_sources.write().await = excluded_sources;
+
+        let mut stacks = self.stacks.write().await;
+        for stack in stacks.values_mut() {
+            stack.data.prune_by_excluded_sources(&exclusion_set);
```

Comment:
❔Maybe a filter function on the stack struct is a bit cleaner, or we just reset the stack like we do in setMarket